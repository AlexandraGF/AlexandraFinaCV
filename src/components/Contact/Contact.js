import React, { Component } from 'react';
import './Contact.scss';

class Contact extends Component {
    constructor(props) {
        super(props);
        this.state = {clicked: false};
    
        this.handleClick = this.handleClick.bind(this);
      }

      handleClick() {
        this.setState(state => ({
          clicked: !state.clicked
        }))
    };

    render() {
        return (
            <div>
                <div className="contactHeading" onClick={this.handleClick}><div className="contactTitle">Contact Details</div><div className="contactIconImage"> {!this.state.clicked ? "+": "-"}</div></div> 
                    {this.state.clicked ? (
                    <div className="contactBody">
                    <div>58 Cornwall Road</div>
                    <div>London, SE1 9PU</div>
                    <div>Mobile: 07946 749 086</div>
                    <div>Email: alexandra.fina@yahoo.com</div>
                    </div>
                    ) : "" }     
            </div>
        );
    }
}

export default Contact;
