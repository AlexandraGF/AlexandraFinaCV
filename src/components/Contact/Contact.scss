@import '../../variables.scss';

.contactHeading {
    font-family: 'Allura', cursive;
    font-size: 30px;
    margin: 7% 7% 0 11%;
    padding-left: 6%;
    box-shadow: $boxShadow;
    border-radius: $borderRadius;
    display: inline-block;
    width: 80%;
}

.contactTitle {
    width: 80%;
}

.contactIconImage {
    width: 8%;
}

.contactTitle, .contactIconImage {
    flex-wrap: wrap;
    display: inline-flex;
}

.contactHeading:hover {
    transform: $hoverSize;
    cursor: $cursorPointer;
}

.contactBody {
    box-shadow: 5px 10px 12px #a2c3c3;
    margin: 7% 4% 7% 13%;
    font-family: 'K2D', sans-serif;
}
