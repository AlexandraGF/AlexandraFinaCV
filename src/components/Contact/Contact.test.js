import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

import Contact from './Contact';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Contact />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders text', () => {
    const wrapper = mount(<Contact />);
    expect(wrapper.text()).toContain("Contact Details");
    });
    
    it('renders Contact Component', () => {
    const tree = renderer.create(
    <Contact />
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
    })
    