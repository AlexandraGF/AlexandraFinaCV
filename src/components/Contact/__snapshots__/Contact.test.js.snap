// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Contact Component 1`] = `
<div>
  <div
    className="contactHeading"
    onClick={[Function]}
  >
    <div
      className="contactTitle"
    >
      Contact Details
    </div>
    <div
      className="contactIconImage"
    >
       
      <img
        src="plus.png"
      />
    </div>
  </div>
  
</div>
`;
